src/Symfony/Component/ExpressionLanguage/Node/BinaryNode.php

```diff
@@ -52,6 +52,10 @@ public function compile(Compiler $compiler): void
         if ('matches' == $operator) {
             if ($this->nodes['right'] instanceof ConstantNode) {
                 $this->evaluateMatches($this->nodes['right']->evaluate([], []), '');
+            } elseif ($this->nodes['right'] instanceof self) {
+                if ('~' !== $this->nodes['right']->attributes['operator']) {
+                    throw new SyntaxError('A boolean is passed to "matches" instead of a regexp.');
```

Comment:
`..` and `**` produce invalid values for the rehex, but they are not booleans

Reply:
good catch! updated to use `Regexp passed to "matches" is not a string.`